lib/models.dart

```diff
@@ -129,6 +124,43 @@ class Playlist extends Comparable<Playlist> {
 
     return 0;
   }
+
+  List<PlaylistItem>? getEndangeredVideos(Playlist playlist) {
```

Comment:
Ideally models should not have any logic attached to them, they should be dumb and just a property bag. You can move this method to some extension